crates/genesis/src/rollup.rs

```diff
@@ -0,0 +1,708 @@
+    pub granite_time: Option<u64>,
+    /// `holocene_time` sets the activation time for the Holocene network upgrade.
+    /// Active if `holocene_time` != None && L2 block timestamp >= Some(holocene_time), inactive
+    /// otherwise.
+    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
+    pub holocene_time: Option<u64>,
+    /// `batch_inbox_address` is the L1 address that batches are sent to.
+    pub batch_inbox_address: Address,
+    /// `deposit_contract_address` is the L1 address that deposits are sent to.
+    pub deposit_contract_address: Address,
+    /// `l1_system_config_address` is the L1 address that the system config is stored at.
+    pub l1_system_config_address: Address,
+    /// `protocol_versions_address` is the L1 address that the protocol versions are stored at.
+    pub protocol_versions_address: Address,
+    /// The superchain config address.
+    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
+    pub superchain_config_address: Option<Address>,
+    /// `blobs_enabled_l1_timestamp` is the timestamp to start reading blobs as a batch data
+    /// source. Optional.
+    #[cfg_attr(
+        feature = "serde",
+        serde(rename = "blobs_data", skip_serializing_if = "Option::is_none")
+    )]
+    pub blobs_enabled_l1_timestamp: Option<u64>,
+    /// `da_challenge_address` is the L1 address that the data availability challenge contract is
+    /// stored at.
+    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
+    pub da_challenge_address: Option<Address>,
+}
+
+// Need to manually implement Default because [`BaseFeeParams`] has no Default impl.
+impl Default for RollupConfig {
+    fn default() -> Self {
+        RollupConfig {
+            genesis: ChainGenesis::default(),
+            block_time: 0,
+            max_sequencer_drift: 0,
+            seq_window_size: 0,
+            channel_timeout: 0,
+            granite_channel_timeout: GRANITE_CHANNEL_TIMEOUT,
+            l1_chain_id: 0,
+            l2_chain_id: 0,
+            base_fee_params: OP_BASE_FEE_PARAMS,
+            canyon_base_fee_params: OP_CANYON_BASE_FEE_PARAMS,
+            regolith_time: None,
+            canyon_time: None,
+            delta_time: None,
+            ecotone_time: None,
+            fjord_time: None,
+            granite_time: None,
+            holocene_time: None,
+            batch_inbox_address: Address::ZERO,
+            deposit_contract_address: Address::ZERO,
+            l1_system_config_address: Address::ZERO,
+            protocol_versions_address: Address::ZERO,
+            superchain_config_address: None,
+            blobs_enabled_l1_timestamp: None,
+            da_challenge_address: None,
+        }
+    }
+}
+
+/// Loads the rollup config for the OP-Stack chain given the chain config and address list.
+pub fn load_op_stack_rollup_config(chain_config: &ChainConfig) -> RollupConfig {
```

Comment:
can we make this a function of chainconfig instead?

Reply:
Definitely good idea.